src/mmw/js/src/draw/views.js

```diff
@@ -436,10 +436,12 @@ var WatershedDelineationView = Marionette.ItemView.extend({
             .then(function(latlng) {
                 // Assign `rwdClickedPoint` so it's in scope to be removed from
                 // the map when the deferred chain completes.
-                rwdClickedPoint = L.marker(latlng, {
-                    icon: drawUtils.createRwdMarkerIcon('original-point')
-                }).bindPopup('Original clicked outlet point');
-                rwdClickedPoint.addTo(map);
+                if (map) {
```

Comment:
Is this guard on `map` necessary?  In recreating the error, `map` was not undefined and given that it's an argument to the function which creates this promise, this function would never execute.  If `App.getLeafletMap()` is returning a falsey value, then we'd need to add the guard at `utils.placeMarker` which is where an exception would originate.  However, I suspect that we couldn't get to this method at all without a valid map reference.  Consider removing this nesting and guard, or if it is needed, moving it to before the first usage of `map`.